src/gpu/generic/sycl/ref_shuffle.hpp

```diff
@@ -42,18 +42,20 @@ struct ref_shuffle_t : public gpu::generic::sycl::primitive_t {
         status_t init(impl::engine_t *engine) {
             using namespace format_tag;
             using namespace data_type;
-            const memory_desc_wrapper data_d(src_md(0));
-            const memory_desc_wrapper dst_d(dst_md(0));
+            auto src_data_md = is_fwd() ? src_md() : diff_src_md();
+            auto dst_data_md = is_fwd() ? dst_md() : diff_dst_md();
 
-            const bool ok = src_md()->data_type == dst_md()->data_type
-                    && (src_md()->data_type == bf16 || src_md()->data_type == s8
-                            || src_md()->data_type == u8
-                            || src_md()->data_type == f16
-                            || src_md()->data_type == f32)
-                    && (src_md(0)->format_desc.blocking.inner_nblks == 0)
-                    && attr()->has_default_values()
+            const bool ok = src_data_md->data_type == dst_data_md->data_type
+                    && (src_data_md->data_type == bf16
+                            || src_data_md->data_type == f16
+                            || src_data_md->data_type == f32
+                            || (is_fwd() && src_data_md->data_type == s32)
+                            || (is_fwd() && src_data_md->data_type == s8)
+                            || (is_fwd() && src_data_md->data_type == u8))
```

Comment:
There's `utils::one_of(src_data_type, dt0, dt1, ...)` function for such enumeration.

Reply:
Suggestions included thanks.